expression/builtin_time.go

```diff
@@ -1786,7 +1839,51 @@ type builtinAddTimeSig struct {
 // eval evals a builtinAddTimeSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_addtime
 func (b *builtinAddTimeSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("ADDTIME")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, nil
+	}
+	s, err := args[1].ToString()
+	var arg1 types.Duration
+	if getFsp(s) == 0 {
+		arg1, err = types.ParseDuration(s, 0)
+	} else {
+		arg1, err = types.ParseDuration(s, types.MaxFsp)
+	}
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	switch tp := args[0].Kind(); tp {
+	case types.KindMysqlTime:
+		arg0 := args[0].GetMysqlTime()
+		tmpDuration := arg0.Add(arg1)
+		result, err := tmpDuration.ConvertToTime(arg0.Type)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		d.SetMysqlTime(result)
+		return d, nil
+	case types.KindMysqlDuration:
+		arg0 := args[0].GetMysqlDuration()
+		result, err := arg0.Add(arg1)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		d.SetMysqlDuration(result)
+		return d, nil
+	default:
+		ss, err := args[0].ToString()
```

Comment:
because it is hard to complete a function to convert every types directly to Time or Duration.   it is very easy that it is converted  to string and string is converted to Time or Duration.